### DISCUSSION TITLE
Ambient aerosol particles

### DISCUSSION THREAD

**Discussion by maxicloco** [contrailcirrus/pycontrails#137] [Q&A] [answered]
Hello, I am using pycontails to not only simulate contrails due to Jet A-1 or SAF but also liquid hydrogen combustion. Since there are no additional nvPM Emissions due to LH2 the ambient aerosol concentration becomes more important. At the moment a "min_ice_particle_number_nvpm_ei_n" is defined to account for ambient aerosol particles in case the nvPM are below this threshhold. Unfortunatly the ice particle number is thereafter calculated with the fuel per segment length 
("return fuel_dist * np.maximum(nvpm_ei_n_activated, min_ice_particle_number_nvpm_ei_n) * f_surv"). 
This is unproblematic for Jet A-1 and SAF but the fuel consumption for LH2 is only about 30 %.  In order to simulate LH2 it would probably be better to take a ambient aerosol concentration not dependent on the fuel consumption. If I have made a mistake please let me know and thank you in advance! I am using pycontrails V0.49.0

**Comment by thabbott** [comment]
Thanks for starting this discussion, @maxicloco. We're taking a look at this and will follow up soon!

**Comment by thabbott** [comment, marked as answer]
I think you bring up a good point, @maxicloco. `min_ice_particle_number_nvpm_ei_n` is currently used to represent activation of secondary aerosols produced by engine emissions *and* ambient aerosols:
https://github.com/contrailcirrus/pycontrails/blob/2c352055a19a19a51c45077de2e647319da384d9/pycontrails/models/cocip/contrail_properties.py#L204-L218
Scaling by fuel consumption seems like the right thing to do for secondary aerosols, since they form from combustion byproducts. As you point out, it's probably not the ideal way to represent ambient aerosols.

We could think about separating `min_ice_particle_number_nvpm_ei_n` into two parameters: one that represents activation of secondary aerosols by setting a floor for `nvpm_ei_n`, and a second parameter that enters the ice particle number calculation independently from `fuel_dist` and represents activation of ambient aerosols. The details might be tricky, though. Converting an ambient aerosol number concentration (dimensions of number per volume) to an ice particle number (dimensions of number per distance) would require knowing the plume area at time of nucleation, which isn't something we have to worry about in the current ice particle number calculation.

If we did this, we'd also want to make the floor for `nvpm_ei_n` dependent on fuel type, since only certain fuels produce secondary aerosols.

@zebengberg and @trdean1, what do you think about moving this to an issue? (I would classify it as a feature request rather than a bug fix. Our current approach seems fine for Jet-A and SAF, but figuring out a way to better represent nucleation on ambient aerosols might be more appropriate for LH2.)

**Comment by roger-teoh** [comment]
Hi @maxicloco , thank you for your question. We have previously used pycontrails to perform an exploratory study to evaluate the effects of liquid hydrogen (LH2) on the change in persistent contrail formation, properties, and climate forcing. We had to resolve these similar questions and came up with the following workaround that is consistent with what was proposed by @thabbott : 

- You are right that the lower calorific value of LH2 is higher than Jet A, and therefore, the fuel consumption per flight distance flown should be smaller than conventional fuels. We assume that the fuel consumption for LH2 scales as a ratio of the lower calorific value of the Jet A divided by the LH2 fuel. 
- As we do not know the exact ambient aerosol concentration, we reverted to a sensitivity analysis and assume a fixed "nvPM" EIn of 10^13, 10^14, and 10^15 particles/kg at all flight waypoints to approximate for the potential differences in the activation of ambient aerosols. This EIn range was informed by Fig. 3 of **Karcher et al. (2018) Formation and radiative forcing of contrail cirrus**

We acknowledge that this solution is not ideal, but it should provide a quick workaround to the issues you are facing. We agree that the global spatiotemporal variability in ambient aerosol concentration is important for the simulation of LH2, and are currently working on incorporating this feature into pycontrails. Depending on development priorities, we expect this feature to be integrated in pycontrails within two months.

**Comment by maxicloco** [comment]
Hello @thabbott and @roger-teoh , thanks to both of you for the detailed updates and information! Thanks @roger-teoh for your solution for the problem. I am really looking forward to the coming updates! I would end the discussion as resolved.

**Comment by JSA7531** [comment]
Hello folks,

Thank you all for this discussion on modeling the effects of Hydrogen combustion and ambient particle contributions. As I also have an application for Hydrogen combustion, I wanted to following with some additional questions:

1. First, thanks to Dr. Teoh for the suggested ranges of nvPM EIn values. If there is an updated suggestion, I would appreciate it if you could let me know (or point me to the right resource).
2. Can you please point me to where in PyContrails I can modify the H2O emissions index so that it is aligned with the characteristics of my specific application?
3. Can I assume that the following performance parameters can be specified to reflect Hydrogen as the fuel, and no further "adjustments" are necessary? - e.g., fuel flow rate of the Hydrogen fuel with no adjustments. 
- Fuel Flow
- Net Thrust
- Engine Efficiency

Thank you

**Reply by zebengberg** [reply]
Hi @JSA7531 

@roger-teoh can provide some guidance here. pycontrails does support creation of arbitrary fuel types -- see the [fuel module](https://github.com/contrailcirrus/pycontrails/blob/main/pycontrails/core/fuel.py), which already includes a `HydrogenFuel` parameterization. You could create your own `Fuel` instance, or copy and modify the existing `HydrogenFuel`.

**Reply by JSA7531** [reply]
Thank you for the response. I will use the fuel module accordingly to model hydrogen fuel.

And I will also use the suggested sensitivity analysis approach and fix the nvPM EIn to 10^13, 10^14, and 10^15 particles/kg to simulate ambient particles.

Has there been any further work on this:

_"We could think about separating min_ice_particle_number_nvpm_ei_n into two parameters: one that represents activation of secondary aerosols by setting a floor for nvpm_ei_n, and a second parameter that enters the ice particle number calculation independently from fuel_dist and represents activation of ambient aerosols."_

Any other suggestions would be appreciated.

Thank you

**Comment by JSA7531** [comment]
Hello,

I have a point of clarification that I would like to have resolved in this thread.

The default value for min_ice_particle_number_nvpm_ei_n seems to be 10^13 #/kg to account for secondary aerosols and activation of background particles. Roger Teoh suggested a range of values as a sensitivity study. In any case, this value is "normally" applied to Jet A fuel flow.

If modeling hydrogen combustion, would this same 10^13 #/kg value be applied to the much lower hydrogen fuel flow (approximately 35% that of Jet A), thus producing less number of particles? 

If the use of this 10^13 #/kg value is meant as a proxy for the activation of entrained ambient particles, shouldn't the number of particles be equal whether you use Jet A or Hydrogen? Therefore, for hydrogen combustion, should this minimum EI be increased by the ratio of fuel flows (Jet A to hydrogen) to account for the lower hydrogen fuel flow?

I keep thinking I'm misunderstanding something. I'd appreciate any clarifications/guidance you can provide.

Thank you
